pkg/controller/mobilesecurityserviceapp/controller.go

```diff
@@ -133,25 +133,42 @@ func (r *ReconcileMobileSecurityServiceApp) Reconcile(request reconcile.Request)
 
 	reqLogger.Info("Checking for service instance ...")
 	mssInstance := &mobilesecurityservicev1alpha1.MobileSecurityService{}
+
 	operatorNamespace, err := k8sutil.GetOperatorNamespace()
 
 	// Check if it is a local env or an unit test
 	if err == k8sutil.ErrNoNamespace {
-		operatorNamespace = utils.OPERATOR_NAMESPACE_FOR_LOCAL_ENV
+		operatorNamespace = utils.OperatorNamespaceForLocalEnv
 	}
 
-	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: "mobile-security-service", Namespace: operatorNamespace}, mssInstance); err != nil {
-		// Return and don't create
-		reqLogger.Info("Mobile Security Service instance resource not found. Mobile Security Service Application is required to create the application")
-		return reconcile.Result{}, nil
-	}
+	// GET MSS CR
+	r.client.Get(context.TODO(), types.NamespacedName{Name: utils.MobileSecurityServiceCRName, Namespace: operatorNamespace}, mssInstance)
 
 	//Get the REST Service Endpoint
 	serviceAPI := service.GetServiceAPIURL(mssInstance)
 
 	//Check if the APP CR was marked to be deleted
 	isAppMarkedToBeDeleted := instance.GetDeletionTimestamp() != nil
 	if isAppMarkedToBeDeleted {
+
+		// If the Service was deleted and/or marked to be deleted
+		if err := r.client.Get(context.TODO(), types.NamespacedName{Name: utils.MobileSecurityServiceCRName, Namespace: operatorNamespace}, mssInstance); err != nil || mssInstance.GetDeletionTimestamp() != nil {
```

Comment:
I could be wrong but the logic that occurs inside this condition is just removing the finalizers. So you could add a condition to only update if there are finalizers to delete. 

```suggestion
if err := r.client.Get(context.TODO(), types.NamespacedName{Name: utils.MobileSecurityServiceCRName, Namespace: operatorNamespace}, mssInstance); (err != nil || mssInstance.GetDeletionTimestamp() != nil) && len(instance.GetFinalizers()) > 0 {
```